doc/examples/example_asymm_crypto.md

```diff
@@ -298,28 +310,31 @@ The default values for these options are:
 * Salt Length: Length of the digests produced by the hash function used to create the signature digests
 
 A PSS Padding with default options can be created as shown below:
+
 ```cpp
 std::shared_ptr<RSASignaturePadding> padding = std::make_shared<PSSPadding>();
 ```
 
-The following example shows how to change the hash function (e.g. to SHA512) used by the Mask Generation Function. Please note that currently only
+The following example shows how to change the hash function (e.g. to SHA512) used by the Mask Generation Function. Currently, only
 MGF1 is available:
+
 ```cpp
 std::shared_ptr<MaskGenerationFunction> mgf = std::make_shared<MGF1>(mococrw::openssl::DigestTypes::SHA512);
 
 std::shared_ptr<RSASignaturePadding> padding = std::make_shared<PSSPadding>(mgf);
 ```
 
-Please note that enforces the hash function of MGF1 to be SHA512, regardless of the hash function used to create the signed hash.
+Please note the enforcing of the hash function of MGF1 to be SHA512, regardless of the hash function used to create the signed hash.
+
+If the salt length is a non-default value, `nullptr` can be passed as padding, which keeps the default behaviour of the Mask Generation Function. The following example sets the salt length to 20, while keeping the default behaviour for the Mask Generation Function:
 
-If the salt length shall be a non-default value, `nullptr` can be passed as padding which keeps the default behaviour of the Mask Generation Function. The following example sets the salt length to 20, while keeping the default behaviour for the Mask Generation Function:
 ```cpp
 std::shared_ptr<RSASignaturePadding> padding = std::make_shared<PSSPadding>(nullptr, 20);
 ```
 
 # ECDSA
 
-ECDSA only allows to choose the hash function that is used to create the signed hash. The default value is `SHA256`.
+ECDSA only allows the choice of hash function used to create the signed hash. The default value is `SHA256`.
```

Comment:
really? Sounds a bit odd to me

Reply:
Generally, "allows" sounds better when it follows an application or individual. See: https://english.stackexchange.com/questions/85069/is-the-construction-it-allows-to-proper-english

Reply:
It's not about the allows but rather the "choice of hash function" part that confuses me

Reply:
I think it is best I re-word the entire sentence rather than trying to keep it as close to the original, in this case.

Reply:
Done.